src/Core/Request/Customers/CustomerLoginRequest.php

```diff
@@ -67,16 +69,19 @@ class CustomerLoginRequest extends AbstractApiRequest
     /**
      * @param string $email
      * @param string $password
-     * @param string $anonymousCartId
+     * @param string $anonymousCart
      * @param Context $context
      */
-    public function __construct($email, $password, $anonymousCartId = null, $anonymousCart = null, Context $context = null)
+    public function __construct($email, $password, $anonymousCart = null, Context $context = null)
     {
         parent::__construct(LoginEndpoint::endpoint(), $context);
         $this->email = $email;
         $this->password = $password;
-        $this->anonymousCartId = $anonymousCartId;
-        $this->anonymousCart = $anonymousCart;
+        if ($anonymousCart instanceof CartReference) {
+            $this->anonymousCart = $anonymousCart;
+        } else {
+            $this->anonymousCartId = $anonymousCart;
```

Comment:
You could create a CartReference here

```suggestion
            $this->anonymousCart = CartReference::ofId($anonymousCart);
```